### PR TITLE
chore(ci): Remove Docker Compose and run package builds inside our CI image

### LINKED ISSUE #3835: Fully remove `docker-compose.yml`
If I understand correctly, we are working to simplify our environment by shifting logic into the Makefile and away from `scripts/*` and `docker-compose.yml`. We still have a Makefile command called `RUN` that executes a specified `docker-compose.yml` job. We should remove this in favor of using `CONTAINER_TOOL` directly in the Makefile.

### PR DESCRIPTION
**Pull request by jamtur01**
This PR does not very much but removes a lot of code.

1. We build our Deb and RPM packages inside the ci_image
2. We remove all of the verifier jobs from Docker Compose (c.f. #4548)
3. We remove the builder jobs from Docker-Compose.
4. We remove the remaining unused Docker Compose jobs (we may replace the `target-graph` job although I can find no one who actually uses it).
5. We tidy up the Makefile to reflect this removal.

Closes #3835 

Signed-off-by: James Turnbull <james@lovedthanlost.net>

### CODE REVIEW

**Review by binarylogic — APPROVED** [review]
Big moves

**Review by Hoverbear — APPROVED** [review]

File: scripts/package-rpm.sh [hunk 1]
```diff
@@ -29,15 +29,6 @@ PACKAGE_VERSION="$("$PROJECT_ROOT/scripts/version.sh")"
...
 echo "TARGET: $TARGET"
 
-#
-# Safeguard
-#
-
-if [[ "$UID" == "0" ]]; then
-  echo "Error: aborting RPM build due to execution as root" >&2
-  exit 1
-fi
```
Comment (MOZGIII) [inline scripts/package-rpm.sh:39]: I'm pretty sure this was here for a reason... :smile: Is this intended, and not something that slipped in after, say, temporary local changes, etc?

**Review by MOZGIII — APPROVED** [review]
Looks good!

File: scripts/package-rpm.sh (hunk 1 above)
Comment (jamtur01) [reply scripts/package-rpm.sh:39]: The restriction on running RPM build as root (damage to the underlying system) isn't relevant in a container world.

**Pull request merged**